### DISCUSSION TITLE
Integrating 3rd Party Auth like Auth0

### DISCUSSION THREAD

**Discussion by ChangoMan** [supabase/supabase#22857] [Questions]
Hello! I am curious if there are any updated guides or tutorials that can help integrate 3rd party auth like Auth0. I've found a few guides but they are pretty old, but are they still relevant?

Integrating Auth0:
https://supabase.com/partners/integrations/auth0

Integrating a Web3 based Auth Provider:
https://docs.moralis.io/authentication-api/evm/integrations/supabase-nodejs

It seems like the steps are similar:
- You get the JWT from the 3rd party auth provider
- Sign it with the JWT secret from Supabase
- You can then get an `access_token` back from Supabase
- Use the `access_token` in the `Authorization` header
- Must be using `{ createClient } from '@supabase/supabase-js'`

I will be using Next.js and the App router, so is there a better way to do this with the `{ createServerClient } from "@supabase/ssr"`?

I would also like to add a row to the Supabase database when a user signs up with the 3rd party auth, is using the `SERVICE_ROLE` the best way to do this?

Thank you!

**Comment by vz443** [comment]
https://github.com/supabase/supabase/blob/master/apps/docs/content/guides/auth/third-party/auth0.mdx

This worked well for me
